docs/tutorial.md

```diff
@@ -0,0 +1,183 @@
+# Oria Operator Tutorial
+
+### An in-depth walkthrough of building and running a oria-operator with memcached-operator.
+
+## Prerequisites
+
+- [Operator SDK](https://sdk.operatorframework.io/docs/installation/) v1.8.0 or newer
+- User authorized with `cluster-admin` permissions.
+- [GNU Make](https://www.gnu.org/software/make/)
+
+## Overview
+
+We will create a sample project to let you know how it works and this sample will:
+
+- Create a Memcached-Operator Deployment using this [tutorial](https://sdk.operatorframework.io/docs/building-operators/golang/tutorial/)
+- Generate the manifests using `config/default`
+- Run the Operator and after success, delete the `ClusterRole` and `ClusterRoleBinding`. Now, the `memcached-operator` will start complaining about it.
+- Create similar ClusterRole template in ScopeTemplate and assciate it with ScopeInstance
+- Run the `oria-operator`, it will create `ClusterRole` and `ClusterRoleBinding`. This will make sure that `memcached-operator` will not complain about RBAC's. 
+
+### Create a new project
+
+Use the [Operator SDK](https://sdk.operatorframework.io/docs/installation/) CLI to create a new memcached-operator from the [tutorial](https://sdk.operatorframework.io/docs/building-operators/golang/tutorial/).
+
+
+```sh
+mkdir -p $HOME/projects/memcached-operator
+cd $HOME/projects/memcached-operator
+# we'll use a domain of example.com
+# so all API groups will be <group>.example.com
+operator-sdk init --domain example.com --repo github.com/example/memcached-operator
+```
+
+### Create a new API and Controller 
+
+```sh
+$ operator-sdk create api --group cache --version v1alpha1 --kind Memcached --resource --controller
+Writing scaffold for you to edit...
+api/v1alpha1/memcached_types.go
+controllers/memcached_controller.go
+...
+```
+
+Define the API for the Memcached Custom Resource(CR) by modifying the Go type definitions at `api/v1alpha1/memcached_types.go` to have the following spec and status:
+
+```
+// MemcachedSpec defines the desired state of Memcached
+type MemcachedSpec struct {
+	//+kubebuilder:validation:Minimum=0
+	// Size is the size of the memcached deployment
+	Size int32 `json:"size"`
+}
+
+// MemcachedStatus defines the observed state of Memcached
+type MemcachedStatus struct {
+	// Nodes are the names of the memcached pods
+	Nodes []string `json:"nodes"`
+}
+```
+
+### Now create docker image and push to the dockerhub
+
+Change below code in Makefile
+
+```
+-IMG ?= controller:latest
++IMG ?= $(IMAGE_TAG_BASE):$(VERSION)
+```
+
+Once done, you do not have to set IMG or any other image variable in the CLI. The following command will build and push an operator image tagged as example.com/memcached-operator:v0.0.1 to Docker Hub:
+
+```
+make docker-build docker-push
```

Comment:
Yeah without the instructions for setting `IMAGE_TAG_BASE` and `VERSION` this `docker-*` command will push to empty.

Reply:
Added comment at top same like go-lang tutorial